PT-2401 - remove check to fix the 'table does not exist' problem when lower_case_table_names=2 is set

ehooi opened the pull request:
pt-online-schema-change failed with 'table `TableName` does not exist', but the table exists.

It seems that #516 introduces this issue.

The variable $tbl changed to lowercase as `tablename` but MySQL returned the original case `TableName` when `lower_case_table_names=2` is set.
As a result, the check that I removed fails.

Since the check appeared to be primarily preventive, I decided to remove it.

it-percona-cla commented:
[![CLA assistant check](https://cla-assistant.percona.com:443/pull/badge/signed)](https://cla-assistant.percona.com:443/percona/percona-toolkit?pullRequest=891) <br/>All committers have signed the CLA.

svetasmirnova reviewed (changes requested):
Please add a test case. You can take t/pt-online-schema-change/case-insensitive.t as example.

ehooi committed:
remove check to fix the 'table does not exist' problem when lower_case_table_names=2 is set

svetasmirnova committed:
PT-2401 - pt-online-schema-change 'table does not exist' on macos

- Modified code, so it still checks table name but based on the lower_case_table_names option

svetasmirnova committed:
Merge branch '3.x' into HEAD

svetasmirnova commented:
I modified the code so it still checks the table name but now takes into account the value of the lower_case_table_names option. I hope this is OK for you. Regarding the test case, I think t/pt-online-schema-change/case-insensitive.t is enough because we cannot modify this test to check the lower_case_table_names option on Linux anyway.
